### DISCUSSION TITLE
cli: Prevent overwriting typst files

### DISCUSSION THREAD

**Discussion by naim94a** [typst/typst#1375] [Ideas]
The other day I worked a few hours on editing a few typst files. 
At the end, I created a Makefile and by mistake executed `typst compile $^` - which cause doc1.typ's pdf to be named doc2.typ (doc2 is now lost). 

I thought it might be a good idea to detect if the output path exists, and it's format doesn't match - typst should bail.

I would be happy to implement this myself.

**Comment by laurmaedje** [comment]
Feel free to make a PR!

**Reply by laurmaedje** [reply]
Based on the objections by @alerque, perhaps we should discuss this a bit more (here or on Discord), before someone starts work on an implementation.

**Comment by alerque** [comment]
No. Respectfully I would suggest this should not be changed. If I call a CLI program with explicit arguments including an input and output name then I expect it to obey. It becomes very obnoxious to deal with programs that do too much to try to "protect" users by guessing what they might have done wrong. The calling syntax might just as well have been *correct*, Typst really has no way to know. Perhaps the previous invocation resulted in a corrupt file. If what it tries to do and when it prompts/daesn't prompt/refuses to do something changes based on some limited set of expectations and circumstantial guesses, it becomes quite hard to script and integrate in other environments.

Next time use `typst compile $< $@` in your Makefile.
